llvm-external-projects/iree-dialects/lib/Dialect/LinalgTransform/Passes/TransformInterpreterPassBase.cpp

```diff
@@ -315,6 +316,24 @@ static void performOptionalDebugActions(
   if (!hasDebugFlags)
     return;
 
+  // Guard debug output stream by a mutex as the pass may be executing in
+  // parallel on multiple ops.
+  static llvm::sys::SmartMutex<true> dbgsMutex;
+  {
+    llvm::sys::SmartScopedLock<true> lock(dbgsMutex);
+    if (target->getContext()->isMultithreadingEnabled()) {
+      // The code above checked that we are in one of the debug types we are
+      // interested in, no need to wrap this in LLVM_DEBUG.
+      llvm::dbgs() << "===== WARNING =====\n"
```

Comment:
frankly .. just `signalPassFailure` .. I fear that with in some form or another this message will be drowned in other noise and we'll waste cycles figuring things out until a forehead slap moment